web-back/lore/[name].ts

```diff
@@ -29,17 +29,23 @@ app.put(
         const name = req.parameters.name;
         const description = req.body.description as string;
         const title = req.body.title as string;
+        const color = req.body.color as string;
+        const image = req.body.image as string;
+        const thumb = req.body.thumb as string;
         if (!description || !title) {
-            return jsonResponse(404, {
-                error: "Missing description or title",
+            return jsonResponse(400, {
+                error: "Bad request",
+                message: "Malformed request, missing description or title",
+                statusCode: "400",
+                timestamp: new Date().toISOString()
             });
         }
 
-        const results = await loreRepo.upsertLore(name, title, description);
+        const results = await loreRepo.upsertLore(name, title, description, color, thumb, image);
         if (!results) {
-            return jsonResponse(404, { error: "Upsert failed." });
+            return jsonResponse(400, { error: "Bad request.", message: "Upsert failed, couldn't update or add the lore.", statusCode: 400, timestamp: new Date().toISOString()});
         }
-        return jsonResponse(200, results);
+        return jsonResponse(200, results); // 201 seems to be a better response code for a successful PUT but 200 is ok if lore already exists
```

Comment:
It may not be the case, but I remember the return value of `upsertLore` changing whether we inserted or not a new row. This may be a opportunity to change the HTTP code